app/actions/deployment_create.rb

```diff
@@ -83,9 +83,24 @@ def create(app:, user_audit_info:, message:)
 
         deployment
       rescue RevisionResolver::NoUpdateRollback, Sequel::ValidationFailed, AppStart::InvalidApp => e
-        error = DeploymentCreate::Error.new(e.message)
+        raise handle_deployment_create_error(e, app)
+      end
+
+      def handle_deployment_create_error(e, app)
+        space_quota_errors = [:space_quota_exceeded.to_s, :space_app_instance_limit_exceeded.to_s]
+        org_quota_errors = [:quota_exceeded.to_s, :app_instance_limit_exceeded.to_s]
+        if space_quota_errors.any? { |substring| e.message.include?(substring) }
+          space_error_msg = " for space #{app.space.name}. This space's quota may not be large enough to support rolling deployments or your configured max-in-flight."
+          error = DeploymentCreate::Error.new(e.message + space_error_msg)
+        elsif org_quota_errors.any? { |substring| e.message.include?(substring) }
+          org_error_msg_1 = " for organization #{app.organization.name}. "
+          org_error_msg_2 = "This organization's quota may not be large enough to support rolling deployments or your configured max-in-flight."
+          error = DeploymentCreate::Error.new(e.message + org_error_msg_1 + org_error_msg_2)
```

Comment:
I wonder if it's worth all this extra logic to make the space/org distinction.  What if you checked for any quota error and simply added `One or more quotas may not be large enough to support rolling deployments or your configured max-in-flight.` to then end?